ics_space_amplification_test.py

```diff
@@ -0,0 +1,192 @@
+        max_used_capacity_gb = fs_size_gb - min_available_capacity_gb
+        self.log.debug("The maximum used filesystem capacity of {} for the last {} minutes is: {} GB/ {} GB".format(
+            node.private_ip_address, time_interval_minutes, max_used_capacity_gb, fs_size_gb))
+        return max_used_capacity_gb
+
+    def _get_nodes_space_amplification(self, dict_nodes_initial_capacity, start_time) -> dict:
+        dict_nodes_space_amplification = {}
+        dict_nodes_used_capacity = self._get_nodes_used_capacity()
+        for node in self.db_cluster.nodes:
+            node_max_used_capacity_gb = self._get_max_used_capacity_over_time_gb(node=node,
+                                                                                 start_time=start_time)
+            dict_nodes_space_amplification[node.private_ip_address] = node_max_used_capacity_gb - \
+                dict_nodes_initial_capacity[
+                node.private_ip_address]
+            self.log.info(
+                f"Node {node.private_ip_address} used capacity changed from {dict_nodes_initial_capacity[node.private_ip_address]} to {dict_nodes_used_capacity[node.private_ip_address]}.")
+            self.log.info(f"Space amplification is: {dict_nodes_space_amplification[node.private_ip_address]} GB")
+        return dict_nodes_space_amplification
+
+    def _get_nodes_space_amplification_after_write(self, dict_nodes_initial_capacity, written_data_size_gb,
+                                                   start_time) -> dict:
+        self.log.info(f"Space amplification results after a write of: {written_data_size_gb} are:")
+        return self._get_nodes_space_amplification(dict_nodes_initial_capacity=dict_nodes_initial_capacity,
+                                                   start_time=start_time)
+
+    def _get_nodes_used_capacity(self) -> dict:
+        """
+
+        :rtype: dictionary with capacity per node-ip
+        """
+        dict_nodes_used_capacity = {}
+        for node in self.db_cluster.nodes:
+            dict_nodes_used_capacity[node.private_ip_address] = self._get_used_capacity_gb(node=node)
+        return dict_nodes_used_capacity
+
+    @retrying(n=80, sleep_time=60, allowed_exceptions=(AssertionError,))
+    def wait_no_compactions_running(self):
+        compaction_query = "sum(scylla_compaction_manager_compactions{})"
+        now = time.time()
+        results = self.prometheus_db.query(query=compaction_query, start=now - 60, end=now)
+        self.log.debug(f"scylla_compaction_manager_compactions: {results}")
+        # if all are zeros the result will be False, otherwise there are still compactions
+        if results:
+            assert any([float(v[1]) for v in results[0]["values"]]) is False, \
+                "Waiting until all compactions settle down"
+
+    def test_ics_space_amplification(self):  # pylint: disable=too-many-locals
+        # too-many-branches,too-many-statements
```

Comment:
please add a docstring to explian what this test is trying todo, what is the goals of it
(maybe it's copy-paste from the trello task, but it should be in the code)

Reply:
done